umm-spec-lib/src/cmr/umm_spec/field_update.clj

```diff
@@ -6,11 +6,60 @@
   [cmr.umm-spec.date-util :as date-util]
   [cmr.umm-spec.umm-spec-core :as spec-core]))
 
+(def DataCenterURL
+  "Data center URL's content type."
+  "DataCenterURL")
+
+(def HomePage
+  "Data center URL's type."
+  "HOME PAGE")
+
 (defn field-update-functions
   "Partial update functions for handling specific update cases"
   [umm]
   {[:Instruments] (partial util/update-in-each umm [:Platforms])})
 
+(defn- get-home-page-url
+  "Get data center home page url from a list of related urls."
+  [related-urls]
+  (when (seq related-urls)
+    (first (for [{:keys [URLContentType Type URL]} related-urls
+                 :when (and (= DataCenterURL URLContentType)
+                            (= HomePage Type))]
+             {:URLContentType DataCenterURL
+              :Type HomePage
+              :URL URL})))) 
+
+(defn- home-page-url?
+  "Check if a related url is a data center home page url."
+  [related-url]
+  (let [{:keys [URLContentType Type URL]} related-url]
+    (if (and (= DataCenterURL URLContentType)
+             (= HomePage Type))
+      true
+      false)))
+
+(defn- home-page-url-update
+  "Apply special homepageurl update to data-center using the update-value.
+   1. If the new homepageurl is provided in the update-value.
+      a. If data-center-relatedurls contains homepageurl, update it with the new homepageurl.
+      b. If data-center-relatedurls doesn't contain homepageurl, add the new homepageurl to it.
+      Note: Anything other than the homepageurl in update-value are ignored. 
+   2.  Update the non ContactInformation part for the data center normally(replace or add)."
+  [data-center update-value]
+  (let [update-value-non-contactinfo-part (dissoc update-value :ContactInformation)
+        update-value-relatedurls (get-in update-value [:ContactInformation :RelatedUrls])
+        update-value-homepageurl (get-home-page-url update-value-relatedurls)
+        data-center-relatedurls (get-in data-center [:ContactInformation :RelatedUrls])]
+    (-> (if (seq update-value-homepageurl)
```

Comment:
Thread macros are generally for a series of 3 or more function composition (only 2 here).

Reply:
Done.